src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -254,8 +254,14 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
                         var source = remoteProvider.Source;
                         if (source.IsHttp && !source.IsHttps && !source.AllowInsecureConnections)
                         {
-                            await _logger.LogAsync(RestoreLogMessage.CreateError(NuGetLogCode.NU1302,
+                            if ((_request.Project.RestoreMetadata.SdkAnalysisLevel is not null &&
+                                _request.Project.RestoreMetadata.SdkAnalysisLevel >= new NuGetVersion(Shared.SdkAnalysisLevelMinimums.HttpErrorSdkAnalysisLevelMinimumValue)) ||
+                                (_request.Project.RestoreMetadata.UsingMicrosoftNETSdk == false && _request.Project.RestoreMetadata.SdkAnalysisLevel is null))
```

Comment:
Do we need the `_request.Project.RestoreMetadata.UsingMicrosoftNETSdk == false` check? 

If SdkAnalysisLevel is null, should we just log the error? 
Seems easier to read and yields the same result.

Reply:
I was following this logic https://github.com/NuGet/NuGet.Client/pull/5896#issue-2385635403. But since this is only going into .Net 9 only, we could remove it I guess.  @zivkan what do you think?

Reply:
It's needed, as described here: https://github.com/NuGet/NuGet.Client/blob/dev/docs/feature-guide.md#warnings-and-defaults

SDK and non-SDK projects have **different** defaults.